# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsPIDOMWindow.h
# Commit: f8a2a169b296
# Full Hash: f8a2a169b2969d29e04733ae27a56459e5359973
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Revert "Bug 1550462 - part 3: Enable `pointerrawupdate` event r=smaug" for causing mochitest failures @ test_pointerrawupdate_event_count.html
#   
#   This reverts commit eb27445e0352d6b4df540adb40e21f72fc052e80.
#   
#   Revert "Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar"
# ==============================================================================

diff -r 8d25773fc82f -r f8a2a169b296 dom/base/nsPIDOMWindow.h
--- a/dom/base/nsPIDOMWindow.h	Fri May 16 06:20:47 2025 +0000
+++ b/dom/base/nsPIDOMWindow.h	Fri May 16 09:30:09 2025 +0300
@@ -226,29 +226,6 @@
 
   /**
    * Call this to check whether some node (this window, its document,
-   * or content in that document) has a pointerrawupdate event listener.
-   */
-  bool HasPointerRawUpdateEventListeners() const {
-    return mMayHavePointerRawUpdateEventListener;
-  }
-
-  /**
-   * Call this to indicate that some node (this window, its document,
-   * or content in that document) has a pointerrawupdate event listener.
-   * This may not accept that if the event is not available in this window.
-   */
-  void MaybeSetHasPointerRawUpdateEventListeners();
-
- protected:
-  /**
-   * Call this to clear whether some nodes has a pointerrawupdate event
-   * listener.
-   */
-  void ClearHasPointerRawUpdateEventListeners();
-
- public:
-  /**
-   * Call this to check whether some node (this window, its document,
    * or content in that document) has a transition* event listeners.
    */
   bool HasTransitionEventListeners() { return mMayHaveTransitionEventListener; }
@@ -672,25 +649,24 @@
   RefPtr<mozilla::dom::Navigator> mNavigator;
 
   // These variables are only used on inner windows.
-  uint32_t mMutationBits = 0;
+  uint32_t mMutationBits;
 
   uint32_t mActivePeerConnections = 0;
 
-  bool mIsDocumentLoaded = false;
-  bool mIsHandlingResizeEvent = false;
-  bool mMayHaveDOMActivateEventListeners = false;
-  bool mMayHaveTouchEventListener = false;
-  bool mMayHaveSelectionChangeEventListener = false;
-  bool mMayHaveFormSelectEventListener = false;
-  bool mMayHaveMouseEnterLeaveEventListener = false;
-  bool mMayHavePointerEnterLeaveEventListener = false;
-  bool mMayHavePointerRawUpdateEventListener = false;
-  bool mMayHaveTransitionEventListener = false;
-  bool mMayHaveSMILTimeEventListener = false;
+  bool mIsDocumentLoaded;
+  bool mIsHandlingResizeEvent;
+  bool mMayHaveDOMActivateEventListeners;
+  bool mMayHaveTouchEventListener;
+  bool mMayHaveSelectionChangeEventListener;
+  bool mMayHaveFormSelectEventListener;
+  bool mMayHaveMouseEnterLeaveEventListener;
+  bool mMayHavePointerEnterLeaveEventListener;
+  bool mMayHaveTransitionEventListener;
+  bool mMayHaveSMILTimeEventListener;
   // Only used for telemetry probes.  This may be wrong if some nodes have
   // come from another document with `Document.adoptNode`.
-  bool mMayHaveBeforeInputEventListenerForTelemetry = false;
-  bool mMutationObserverHasObservedNodeForTelemetry = false;
+  bool mMayHaveBeforeInputEventListenerForTelemetry;
+  bool mMutationObserverHasObservedNodeForTelemetry;
 
   // Our inner window's outer window.
   nsCOMPtr<nsPIDOMWindowOuter> mOuterWindow;
@@ -711,11 +687,11 @@
 
   // A unique (as long as our 64-bit counter doesn't roll over) id for
   // this window.
-  uint64_t mWindowID = 0;
+  uint64_t mWindowID;
 
   // Set to true once we've sent the (chrome|content)-document-global-created
   // notification.
-  bool mHasNotifiedGlobalCreated = false;
+  bool mHasNotifiedGlobalCreated;
 
   // Whether when focused via an "unknown" focus method, we should show outlines
   // by default or not. The initial value of this is true (so as to show
@@ -723,7 +699,7 @@
   // without any other user interaction).
   bool mUnknownFocusMethodShouldShowOutline = true;
 
-  uint32_t mMarkedCCGeneration = 0;
+  uint32_t mMarkedCCGeneration;
 
   // mTopInnerWindow is used for tab-wise check by timeout throttling. It could
   // be null.
@@ -732,17 +708,17 @@
   // The evidence that we have tried to cache mTopInnerWindow only once from
   // SetNewDocument(). Note: We need this extra flag because mTopInnerWindow
   // could be null and we don't want it to be set multiple times.
-  bool mHasTriedToCacheTopInnerWindow = false;
+  bool mHasTriedToCacheTopInnerWindow;
 
   // The number of active IndexedDB databases.
-  uint32_t mNumOfIndexedDBDatabases = 0;
+  uint32_t mNumOfIndexedDBDatabases;
 
   // The number of open WebSockets.
-  uint32_t mNumOfOpenWebSockets = 0;
+  uint32_t mNumOfOpenWebSockets;
 
   // The event dispatch code sets and unsets this while keeping
   // the event object alive.
-  mozilla::dom::Event* mEvent = nullptr;
+  mozilla::dom::Event* mEvent;
 
   // The WindowGlobalChild actor for this window.
   //
@@ -750,7 +726,7 @@
   // during SetNewDocument, and cleared during FreeInnerObjects.
   RefPtr<mozilla::dom::WindowGlobalChild> mWindowGlobalChild;
 
-  bool mWasSuspendedByGroup = false;
+  bool mWasSuspendedByGroup;
 
   /**
    * Count of the number of active LockRequest objects, including ones from